src/main/java/ladder/domain/line/Line.java

```diff
@@ -1,60 +1,55 @@
 package ladder.domain.line;
 
-import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.IntStream;
 import ladder.domain.exception.InvalidCountOfPersonException;
+import ladder.domain.exception.InvalidLineException;
+import ladder.domain.exception.MismatchPersonBarsException;
 
-public abstract class Line {
+public class Line {
 
-    protected final List<Boolean> bars;
-    protected final int countOfPerson;
+    private final List<Bar> bars;
+    private final int countOfPerson;
 
-    protected Line(int countOfPerson) {
-        this.bars = new ArrayList<>();
+    public Line(int countOfPerson, List<Bar> bars) {
+        this.bars = bars;
         this.countOfPerson = validateCountOfPerson(countOfPerson);
-        initBars();
+        validateBar();
     }
 
-    public List<Boolean> getBars() {
+    public List<Bar> getBars() {
         return this.bars;
     }
 
-    public int move(int index) {
-        if (isMovementAvailable(index)) {
-            return -1;
-        }
-
-        if (isMovementAvailable(index + 1)) {
-            return 1;
-        }
-
-        return 0;
+    public Bar getBar(int index) {
+        return bars.get(index);
     }
 
-    private Boolean isMovementAvailable(int index) {
+    public boolean isMovementAvailable(int index) {
         try {
-            return this.bars.get(index);
+            return this.bars.get(index).isAvailable();
         } catch (Exception e) {
             return false;
         }
     }
 
-    public boolean validate() {
+    private void validateBar() {
         if (bars.size() != countOfPerson) {
-            return false;
+            throw new MismatchPersonBarsException();
         }
 
-        return IntStream.range(0, bars.size())
-                .noneMatch(i -> isFirstBarInvalid(i) || isBarInvalid(i));
+        if (IntStream.range(0, bars.size())
+                .anyMatch(i -> isFirstBarInvalid(i) || isBarInvalid(i))) {
+            throw new InvalidLineException();
+        }
     }
 
     private boolean isBarInvalid(int index) {
-        return index != 0 && bars.get(index - 1) && bars.get(index);
+        return index != 0 && getBar(index - 1).isAvailable() && getBar(index).isAvailable();
```

Comment:
```suggestion
        return index != 0 
                && getBar(index - 1).isAvailable() 
                && getBar(index).isAvailable();
```
3개 이상 비교할때는 줄바꿈으로 구분해주면 더 읽기 편해져요!